toolchains/java/WORKSPACE

```diff
@@ -1,2 +1,62 @@
-# only temporary for compatibility with `WORKSPACE` setup
-# TODO: remove when migration to `bzlmod` is complete
\ No newline at end of file
+# NOTE: only temporary for compatibility with `WORKSPACE` setup!
+# TODO: remove when migration to `bzlmod` is completelocal_repository
+
+# NOTE: all of the following is copy-pasted from `/WORKSPACE` and reduced
+# to the necessary to make this repository work compatibly in isolation.
+
+# TODO: remove dependency on cumulative repository to be sure there is no
+# hidden entanglement left. use individual toolchains instead.
+local_repository(
+    name = "io_tweag_rules_nixpkgs",
+    path = "../..",
+)
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@io_tweag_rules_nixpkgs//nixpkgs:repositories.bzl", "rules_nixpkgs_dependencies")
+
+rules_nixpkgs_dependencies()
+
+load("@rules_nixpkgs_core//:nixpkgs.bzl", "nixpkgs_git_repository", "nixpkgs_local_repository")
+load("//:java.bzl", "nixpkgs_java_configure")
+load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_cc_configure")
+
+http_archive(
```

Comment:
Perhaps the docs related parts could be factored out into `@rules_nixpkgs_core//docs:repositories.bzl` or something along those lines?